### PR TITLE
Setup the geos python package CI

### PR DESCRIPTION
**Pull request by cssherman**

### CODE REVIEW

**Comment by cssherman** [comment]
@TotoGaz - any thoughts on the CI configuration?  I'm building off of the github template for python packages here.  Do you want to add/remove any python versions?  Also, I've commented out the linting and testing steps for now, as we haven't been enforcing these yet.

File: .github/workflows/python-package.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+
+jobs:
+  build:
+
+    runs-on: ubuntu-22.04
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.9", "3.10", "3.11"]
```
Comment (TotoGaz) [inline .github/workflows/python-package.yml:17]: Is it possible to have python 3.8? I kind of remember there are some differences for type hints starting python 3.9.
Those differences were popping errors on ubuntu 20.04 (IIRC) but the differences are not worth ditching 3.8 from the supported platforms.

File: .github/workflows/python-package.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        package-name: ["geos_ats_package", "geosx_mesh_doctor", "geosx_mesh_tools_package", "geosx_xml_tools_package", "hdf5_wrapper_package", "pygeosx_tools_package", "timehistory_package"]
+
+    steps:
+    - uses: actions/checkout@v3
+    - uses: mpi4py/setup-mpi@v1
```
Comment (TotoGaz) [inline .github/workflows/python-package.yml:22]: Do we want to rely on this or should we prefer installing from `requirements.txt`?

**Review by TotoGaz — APPROVED** [review]

File: .github/workflows/python-package.yml (hunk 2 above)
Comment (cssherman) [reply .github/workflows/python-package.yml:22]: These help to setup the underlying python and mpi environments (needed for mpi4py to not throw an error).  We could probably roll our own, but these actions seem to be quite reliable.

**Pull request merged**